### PR TITLE
Supporting features for Smart Home

### PR DESCRIPTION
**Pull request by garbear**
## Description

This PR contains three "features", which are additions to the code to enable more use cases in the future. The primary use case being Smart Home support in https://github.com/xbmc/xbmc/pull/21183.

[`Peripherals: Allow devices to have a controller profile`](https://github.com/xbmc/xbmc/commit/81d268e7ba2a5d303080e02ae2b691bf2d50e756)

* This commit makes Kodi more aware of its input peripherals. In order to make use of the extended controller API, Mouse and Keyboard controller profiles were imported (and the other two synced). Importing mouse/keyboard was planned anyway, when keyboard/mouse mapping is expanded in the GUI.

[`Input: Expose input frame to higher-level input`](https://github.com/xbmc/xbmc/commit/b016039a6c4a7251a1f4d1e8b6d89db186333f2d)

* This commit adds a callback found in lower-level driver input to higher-level controller input. At the lower level, we group axes into analog sticks. At the higher level, we will be grouping entire controller state into single-frame events.

[`AppParamParser: Preserve command line arguments`](https://github.com/xbmc/xbmc/commit/c6b320565aa6e779ffb1366aec4892761eed7433)

* This commit stores the command line arguments in a service-accessible state. This is useful for any future services (like ROS 2) that are initialized with the command line parameters.

## Motivation and context

Required for https://github.com/xbmc/xbmc/pull/21183.

## How has this been tested?

My computer vision pipeline has been running relatively stable for about 10 months now.

Without the Mouse and Keyboard profiles, the ROS topic is missing data:

```yaml
header:
  stamp:
    sec: 1648117091
    nanosec: 438027612
  frame_id: nas
peripherals:
- type: 2
  name: Keyboard
  address: keyboard
  vendor_id: 0
  product_id: 0
  controller_profile: ''
- type: 3
  name: Mouse
  address: mouse
  vendor_id: 0
  product_id: 0
  controller_profile: ''
```

With the Mouse and Keyboard profiles included, the ROS topic contains the missing data:

```yaml
header:
  stamp:
    sec: 1648117091
    nanosec: 438027612
  frame_id: nas
peripherals:
- type: 2
  name: Keyboard
  address: keyboard
  vendor_id: 0
  product_id: 0
  controller_profile: game.controller.keyboard
- type: 3
  name: Mouse
  address: mouse
  vendor_id: 0
  product_id: 0
  controller_profile: game.controller.mouse
```

## What is the effect on users?

* Enables future use cases like Smart Home support

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

File: addons/game.controller.keyboard/addon.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="game.controller.keyboard"
+        name="IMB Model M Keyboard"
```
Comment (fuzzard) [inline addons/game.controller.keyboard/addon.xml:3]: Typo might want to fix

Comment (garbear) [reply addons/game.controller.keyboard/addon.xml:3]: thanks, sent upstream: https://github.com/kodi-game/controller-topology-project/pull/130

**Comment by garbear** [comment]
Rebased on https://github.com/xbmc/xbmc/pull/21301, thanks @ksooo the patch got a lot smaller

**Comment by garbear** [comment]
Hey cool, not only did the patch get smaller, but @ksooo fixed the only blocker in https://github.com/xbmc/xbmc/pull/21301. I didn't want to touch system manager/broker, but that went away, so IMO this can go in.

The last discussion is if we should ship the mouse/keyboard add-ons.

Mice and keyboards are always attached, so it makes sense that we would ship them eventually. Indeed, they'll be needed to support controllers emulating keyboards/mice and vice versa (not shipping in Nexus). Might as well include them now though to bring attention to their translations. On the downside, they add a couple images of a mouse/keyboard, which take up a MB or two minimized and compressed.

Another downside is that the IDs will be missing in my ROS 2 work, see the **How has this been tested?** section for what this looks like. Not a big deal, add-ons are easier to rebase than code, but a small amount of code (manifest and cmake install) is touched so there's a small maintenance burden as I rebase https://github.com/xbmc/xbmc/pull/21183 if they're not merged.

FWIW, with the add-ons, mice/keyboards can now be mapped in the GUI. Without mouse/keyboard emulation this is pretty pointless, but we can start collecting buttonmaps for when the feature is added.

**Comment by garbear** [comment]
Screenshots:

Mouse and keyboard now appear in the button mapper:

Map controller

Here's the mouse add-on:

Map mouse

Here's the keyboard add-on:

Map keyboard

**Review by gade01 — APPROVED** [review]
All language files looks great.

**Comment by garbear** [comment]
Anyone not OK with adding the two add-ons? Otherwise I'll merge soon.

**Comment by fuzzard** [comment]
No opposition to the addons. Mouse/keyboard support was probably always intended.

Not entirely sure about the appparamparser argument commit. I guess this is a part of the rest of the smarthome stuff you played with. Just not sure its worth bringing in without an actual use intended.

**Comment by garbear** [comment]
Normally we wouldn't bring something in without an immediate use case, but I'm asking for a favor to merge it so that I can keep rebasing smart home with less effort.

**Comment by garbear** [comment]
I re-synced the controller add-ons after some recent language updates. I also added a note to the raw arguments parameter that it's unused but will be useful if/when ROS 2 support is merged.

**Comment by garbear** [comment]
Good time to merge? I just got my Kinect v2 working continuously (turns out those fan vents on the top are there for a reason and should not be fully covered). When this goes in I can rebase and get everything updated again.

**Review by fuzzard — APPROVED** [review]

**Comment by garbear** [comment]
thanks for approving, and anyone comment on adding these two add-ons if there's an issue with size or translations or whatever!

**Pull request merged**